src/main/java/org/apache/flink/state/benchmark/ttl/TtlListStateBenchmark.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.flink.api.common.state.ListStateDescriptor;
+import org.apache.flink.contrib.streaming.state.RocksDBKeyedStateBackend;
+import org.apache.flink.state.benchmark.StateBenchmarkBase;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.apache.flink.state.benchmark.StateBackendBenchmarkUtils.applyToAllKeys;
+import static org.apache.flink.state.benchmark.StateBackendBenchmarkUtils.compactState;
+import static org.apache.flink.state.benchmark.StateBackendBenchmarkUtils.getListState;
+import static org.apache.flink.state.benchmark.StateBenchmarkConstants.listValueCount;
+import static org.apache.flink.state.benchmark.StateBenchmarkConstants.setupKeyCount;
+
+/** Implementation for list state benchmark testing. */
+public class TtlListStateBenchmark extends TtlStateBenchmarkBase {
+    private final String STATE_NAME = "listState";
+    private final ListStateDescriptor<Long> STATE_DESC =
+            configTtl(new ListStateDescriptor<>(STATE_NAME, Long.class));
+    private ListState<Long> listState;
+    private List<Long> dummyLists;
+
+    public static void main(String[] args) throws RunnerException {
+        Options opt =
+                new OptionsBuilder()
+                        .verbosity(VerboseMode.NORMAL)
+                        .include(".*" + TtlListStateBenchmark.class.getCanonicalName() + ".*")
+                        .build();
+
+        new Runner(opt).run();
+    }
+
+    @Setup
+    public void setUp() throws Exception {
+        keyedStateBackend = createKeyedStateBackend();
+        listState = getListState(keyedStateBackend, STATE_DESC);
+        dummyLists = new ArrayList<>(listValueCount);
+        for (int i = 0; i < listValueCount; ++i) {
+            dummyLists.add(random.nextLong());
+        }
+        keyIndex = new AtomicInteger();
+    }
+
+    @Setup(Level.Iteration)
+    public void setUpPerIteration() throws Exception {
+        for (int i = 0; i < setupKeyCount; ++i) {
+            keyedStateBackend.setCurrentKey((long) i);
+            listState.add(random.nextLong());
+        }
+        // make sure only one sst file left, so all get invocation will access this single file,
+        // to prevent the spike caused by different key distribution in multiple sst files,
+        // the more access to the older sst file, the lower throughput will be.
+        if (keyedStateBackend instanceof RocksDBKeyedStateBackend) {
```

Comment:
I noticed that there existed so much duplicated code here, can how reduce this code via Generics or more abstract class?

Reply:
Actually, in order to prevent small changes from affecting the existing test results, I did not use class abstraction purposely. I suggest we introduce a refactor to the existing tests along with the [FLINK-33864](https://issues.apache.org/jira/browse/FLINK-33864), since it can greatly affect test results. WDYT?

Reply:
I'm not sure whether such refactor could impact the performance results significantly, could you please provide some data if possible? If so, I think we can refactor the tests with FLINK-33864.

Reply:
After some try, I found there is no easy way to do so, since the `Ttl*StateBenchmark` extends the `TtlStateBenchmarkBase`, while they cannot extends `*StateBenchmark` in the meantime. So a possible way is to extract an `TestHarness` or `TestContext` to hold the variables. I only do refactor in `valueAdd`, and test it on my local machine. Before:
```
Benchmark                     (backendType)   Mode  Cnt     Score     Error   Units
ValueStateBenchmark.valueAdd           HEAP  thrpt   30  5815.682 ± 122.008  ops/ms
Benchmark                     (backendType)   Mode  Cnt     Score     Error   Units
ValueStateBenchmark.valueAdd           HEAP  thrpt   30  5817.108 ± 162.588  ops/ms
Benchmark                     (backendType)   Mode  Cnt     Score     Error   Units
ValueStateBenchmark.valueAdd           HEAP  thrpt   30  5792.671 ± 132.356  ops/ms
```
And after the refactor:
```
Benchmark                     (backendType)   Mode  Cnt     Score     Error   Units
ValueStateBenchmark.valueAdd           HEAP  thrpt   30  5712.225 ± 133.996  ops/ms
Benchmark                     (backendType)   Mode  Cnt     Score    Error   Units
ValueStateBenchmark.valueAdd           HEAP  thrpt   30  5733.398 ± 77.725  ops/ms
Benchmark                     (backendType)   Mode  Cnt     Score    Error   Units
ValueStateBenchmark.valueAdd           HEAP  thrpt   30  5771.855 ± 80.658  ops/ms
```
As we can see, there is about 1% impact on the performance results.

Reply:
I think the 1% impact is really small compared to the project maintaining. And we can postpone this later in [FLINK-33864](https://issues.apache.org/jira/browse/FLINK-33864).